pkg/controller/ec2/routetable/controller.go

```diff
@@ -237,33 +250,114 @@ func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+				aws.StringValue(ds.NatGatewayID) == rt.NatGatewayID ||
+				aws.StringValue(ds.NetworkInterfaceID) == rt.NetworkInterfaceID ||
+				aws.StringValue(ds.TransitGatewayID) == rt.TransitGatewayID ||
+				aws.StringValue(ds.VpcPeeringConnectionID) == rt.VpcPeeringConnectionID) {
+
+				found = true
+				break
+			}
+		}
+		if !found {
+			_, err := e.client.DeleteRouteRequest(&awsec2.DeleteRouteInput{
+				RouteTableId:             aws.String(tableID),
+				DestinationCidrBlock:     aws.String(rt.DestinationCIDRBlock),
+				DestinationIpv6CidrBlock: aws.String(rt.DestinationIPV6CIDRBlock),
+			}).Send(ctx)
+
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (e *external) createRoutes(ctx context.Context, tableID string, desired []v1alpha5.Route, observed []v1alpha5.RouteState) error { // nolint:gocyclo
 	for _, rt := range desired {
 		isObserved := false
 		for _, ob := range observed {
-			if ob.GatewayID == aws.StringValue(rt.GatewayID) && ob.DestinationCIDRBlock == aws.StringValue(rt.DestinationCIDRBlock) {
+			if ob.DestinationCIDRBlock == aws.StringValue(rt.DestinationCIDRBlock) && (ob.GatewayID == aws.StringValue(rt.GatewayID) ||
+				ob.InstanceID == aws.StringValue(rt.InstanceID) ||
+				ob.LocalGatewayID == aws.StringValue(rt.LocalGatewayID) ||
+				ob.NatGatewayID == aws.StringValue(rt.NatGatewayID) ||
+				ob.NetworkInterfaceID == aws.StringValue(rt.NetworkInterfaceID) ||
+				ob.TransitGatewayID == aws.StringValue(rt.TransitGatewayID) ||
+				ob.VpcPeeringConnectionID == aws.StringValue(rt.VpcPeeringConnectionID)) {
 				isObserved = true
 				break
 			}
 		}
 		// if the route is already created, skip it
 		if !isObserved {
 			_, err := e.client.CreateRouteRequest(&awsec2.CreateRouteInput{
-				RouteTableId:         aws.String(tableID),
-				DestinationCidrBlock: rt.DestinationCIDRBlock,
-				GatewayId:            rt.GatewayID,
+				RouteTableId:             aws.String(tableID),
+				DestinationCidrBlock:     rt.DestinationCIDRBlock,
+				DestinationIpv6CidrBlock: rt.DestinationIPV6CIDRBlock,
+				GatewayId:                rt.GatewayID,
+				InstanceId:               rt.InstanceID,
+				LocalGatewayId:           rt.LocalGatewayID,
+				NatGatewayId:             rt.NatGatewayID,
+				NetworkInterfaceId:       rt.NetworkInterfaceID,
+				TransitGatewayId:         rt.TransitGatewayID,
+				VpcPeeringConnectionId:   rt.VpcPeeringConnectionID,
 			}).Send(ctx)
 
 			if err != nil {
-				return errors.Wrap(err, errCreateRoute)
+				return err
 			}
 		}
 	}
+	return nil
+}
+
+func (e *external) reconcileRoutes(ctx context.Context, tableID string, desired []v1alpha5.Route, observed []v1alpha5.RouteState) error {
+
+	err := e.deleteRoutes(ctx, tableID, desired, observed)
+
+	if err != nil {
+		return errors.Wrap(err, errDeleteRoute)
+	}
+	err = e.createRoutes(ctx, tableID, desired, observed)
+
+	if err != nil {
+		return errors.Wrap(err, errCreateRoute)
+	}
+	return nil
```

Comment:
```suggestion
	if err := e.deleteRoutes(ctx, tableID, desired, observed); err != nil {
		return errors.Wrap(err, errDeleteRoute)
	}
	return errors.Wrap(e.createRoutes(ctx, tableID, desired, observed), errCreateRoute)
```

Nit: This and the equivalent associations method below could be simplified to the above. `errors.Wrap` returns `nil` if the supplied error is `nil`.

Reply:
Think I got em all